hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreTest.java

```diff
@@ -42,9 +44,21 @@ public abstract class HiveMetastoreTest {
 
   @BeforeClass
   public static void startMetastore() throws Exception {
+    startMetastore(Collections.emptyMap());
+  }
+  public static void startMetastore(Map<String, String> hiveConfOverride) throws Exception {
     HiveMetastoreTest.metastore = new TestHiveMetastore();
-    metastore.start();
-    HiveMetastoreTest.hiveConf = metastore.hiveConf();
+    if (HiveMetastoreTest.hiveConf != null) {
+      if (hiveConfOverride != null) {
+        for (Map.Entry<String, String> kv : hiveConfOverride.entrySet()) {
+          hiveConf.set(kv.getKey(), kv.getValue());
+        }
+      }
+      metastore.start(HiveMetastoreTest.hiveConf);
+    } else {
+      metastore.start();
```

Comment:
We still would like to apply the `hiveConfOverride` here.
I would guess that it would be easier to read if we calculate the `hiveConf` first, and then start the metastore with the final conf